docs/contracts/using-for.rst

```diff
@@ -20,7 +22,15 @@ The first part, ``A``, can be one of:
   only those functions will be attached to the type as member functions.
   Note that private library functions can only be specified when ``using for`` is inside the library.
 - The name of a library (e.g. ``using L for uint;``) -
-  all non-private functions of the library are attached to the type.
+  all non-private functions of the library are attached to the type
+  as member functions
+- a list of assignments of pure file-level or public/internal/private library functions to operators
```

Comment:
This kind of suggests that it can *either* be a list of functions assigned to operators, *or* a list of member functions, while a combination of both seems possible as well (``attaching_and_defining_operator_with_same_function.sol`` is the only test case for that, but it is a test case for that)

By the way: how does the case in that test case occur in the ast? With the same function name twice once with an additional ``operator`` field, once without?

Reply:
Very good point. I have rewritten this bit to make it clearer. I also added a new test case explicitly checking that you can mix defining operators and attaching functions.

Reply:
> By the way: how does the case in that test case occur in the ast? With the same function name twice once with an additional `operator` field, once without?

Yes, exactly:

```solidity
using {add as +, add} for Int;
```
```json
            "functionList":
            [
                {
                    "function":
                    {
                        "id": 3,
                        "name": "add",
                        "nameLocations":
                        [
                            "27:3:0"
                        ],
                        "nodeType": "IdentifierPath",
                        "referencedDeclaration": 33,
                        "src": "27:3:0"
                    },
                    "operator": "+"
                },
                {
                    "function":
                    {
                        "id": 4,
                        "name": "add",
                        "nameLocations":
                        [
                            "37:3:0"
                        ],
                        "nodeType": "IdentifierPath",
                        "referencedDeclaration": 33,
                        "src": "37:3:0"
                    }
                }
            ],
```

Reply:
Making it top-level `` "operator": { "function": { /* as now */ }, "operation": "+" }`` would be an alternative without that problem, but not sure if that's actually better or worse...

Reply:
Changed to `{"operator": "+", "definition": { /* as now */ }}` for operators.